field/TinyLang/Field/F4913.hs

```diff
@@ -0,0 +1,91 @@
+module TinyLang.Field.F4913
+    ( F4913
+    ) where
+
+import           TinyLang.Field.Core
+import           TinyLang.Field.ParsableField
+import           TinyLang.Field.ParserUtils
+import           TinyLang.Prelude
+
+import           Test.QuickCheck
+import qualified Text.Megaparsec.Char.Lexer   as L
+
+{- | A crude implementation of the field of 4193 = 17^3 elements, ie
+     GF(17^3).  We construct F_4193 as F_17[x]/(p), where p is the
+     irreducible polynomial x^3 + x + 3.  More concretely, we define
+     an equivalence relation on the polynomial ring F_17[x] by saying
+     that two polynomials are equivalent if they differ by a multiple
+     of x^3+x+3, and F_4193 is the quotient of the polynomial ring by
+     this relation.  Every element of F_4193 then has a canonical
+     representation as a polynomial a+bx+cx^2.  We add and subtract
+     two such polynomials by adding or subtracting corresponding
+     coefficients (in F_17, so reducing modulo 17 if we're using
+     ints).  To find the product, we multiply polynomials as usual and
+     then use the fact that x^3 = -x-3 to dispose of terms of degree
+     greater than 2: this is hard-coded into the `mul` method.  To
+     find inverses we use the fact that in this field the inverse of
+     any nonzero element a is a^4911 (ie, a^(17^3-2)).
+     We represent a canonical polynomial a+bx+cx^2 using the
+     constructor F a b c with a,b, and c Ints.  Strictly We should use
+     elements of F17, but that would complicate things for little
+     gain.  The concrete syntax for F a b c is <a,b,c>.
```

Comment:
I've changed the syntax from `[a,b,c]` to `<a,b,c>` to avoid clashing with the list syntax (I can change it back if preferable).